include/alpaka/example/ExecuteForEachAccTag.hpp

```diff
@@ -0,0 +1,27 @@
+/* Copyright 2023 Jeffrey Kelling, Bernhard Manfred Gruber, Jan Stephan, Aurora Perego, Andrea Bocci
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#include "alpaka/alpaka.hpp"
+
+#include <functional>
+#include <tuple>
+#include <utility>
+
+#pragma once
+
+namespace alpaka
+{
+    //! execute a callable for each active accelerator tag
+    //
+    // @param callable callable which can be invoked with an accelerator tag
+    // @return disjunction of all invocation results
+    //
+    template<typename TCallable>
+    inline auto executeForEachAccTag(TCallable&& callable)
+    {
+        // Execute the callable once for each enabled accelerator.
+        // Pass the tag as first argument to the callable.
+        return std::apply([=](auto const&... tags) { return (callable(tags) || ...); }, alpaka::EnabledAccTags{});
```

Comment:
@psychocoderHPC sorry, I hadn't thought about this before: does the `||` operator obey the usual short-circuit rules here ?

That is, as soon as one of the `callable` returns `true`, will the rest be skipped ?

Reply:
OK, so we don't want to continue running with the other tags if one fail ?

Reply:
Yes exact, I think we should not waiste compute time if we failed. 
I failure can also bring the device into a state that all followup work fails, in that case the last error will only be an artefact of the failing first run but the user will first look to the last failure.